CRM/Contribute/Form/Task/PDFLetterCommon.php

```diff
@@ -247,13 +247,13 @@ public static function buildContributionArray($groupBy, $contributionIDs, $retur
     $contributions = $contacts = $notSent = array();
     foreach ($contributionIDs as $item => $contributionId) {
       // get contribution information
-      $contribution = CRM_Utils_Token::getContributionTokenDetails(array('contribution_id' => $contributionId),
-        $returnProperties,
-        NULL,
-        $messageToken,
-        $task
-      );
-      $contribution = $contributions[$contributionId] = $contribution[$contributionId];
+
+      // retrieve contribution tokens listed in $messageToken using Contribution.Get API
+      $contribution = civicrm_api3('Contribution', 'getsingle', array(
+        'id' => $contributionId,
+        'return' => CRM_Utils_Array::value('contribution', $messageToken, array()),
```

Comment:
If `$messageToken['contribution']` is empty then this api call will return nothing. Maybe we should wrap the whole thing in a conditional and skip ti altogether in that case?